.github/workflows/behat-test.yml

```diff
@@ -158,12 +158,35 @@ jobs:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
       - name: Download Artifacts
-        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        uses: actions/download-artifact@f44cd7b40bfd40b6aa1cc1b9b5b7bf03d3c67110 # v4.1.0
         with:
-          name: ${{ inputs.name }}-${{ inputs.os }}-test-reports
+          pattern: ${{ inputs.name }}-${{ inputs.os }}-test-reports-*
           path: ${{ inputs.name }}-xunit-reports
+          merge-multiple: true
 
       - uses: ./.github/actions/publish-report
         with:
           check_name: ${{ inputs.name }}-tests-report
-          path: "${{ inputs.name }}-xunit-reports/**/*.xml"
+          path: '${{ inputs.name }}-xunit-reports/**/*.xml'
+
+  regroup-artifacts:
+    needs: [behat-test-run, behat-test-report]
+    if: always()
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        type_of_report: [test-logs, test-reports]
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Merging Artifacts
+        uses: ./.github/actions/merge-artifacts
+        with:
+          artifact_name: ${{ inputs.name }}-${{ inputs.os }}-${{ matrix.type_of_report }}
+          artifact_files: |
+            ${{ inputs.name }}-${{ inputs.os }}-${{ matrix.type_of_report }}/**/*.xml
+            ${{ inputs.name }}-${{ inputs.os }}-${{ matrix.type_of_report }}/**/*.txt
+          artifact_pattern: ${{ inputs.name }}-${{ inputs.os }}-${{ matrix.type_of_report }}-
+          condition: ${{ contains(needs.behat-test-run.result, 'failure') }}
```

Comment:
why do you send condition instead of directly skip the job ?

Reply:
it depends on the matrix's value, you can understand more if you check  https://github.com/centreon/centreon/blob/f41c30f7608a370d494aec5546371a8ea27649d7/.github/workflows/cypress-component-parallelization.yml#L173